src/libexpr/lexer.l

```diff
@@ -12,6 +12,8 @@
 
 
 %{
+#pragma clang diagnostic ignored "-Wunneeded-internal-declaration"
```

Comment:
Note that this causes a gcc warning:
```
src/libexpr/lexer.l:15: warning: ignoring #pragma clang diagnostic [-Wunknown-pragmas]
   15 | #pragma clang diagnostic ignored "-Wunneeded-internal-declaration"
      | 
```

Reply:
Obviously :unamused: 

I've opened https://github.com/NixOS/nix/pull/4300 to fix that one